core/src/tree/TreeTrainer.cpp

```diff
@@ -203,7 +203,6 @@ bool TreeTrainer::split_node_internal(size_t node,
     return true;
   }
 
-  std::fill(responses_by_sample.begin(), responses_by_sample.end(), 0.0);
```

Comment:
I think it'd be good to add a note to `RelabelingStrategy.h` noting that `responses_by_sample` can contain garbage values for indices outside of the given set of sample IDs. This helps define the method contract (which might be surprising to a newcomer).